src/GraphQL/Types/Scalars/DateTimeOffsetGraphType.cs

```diff
@@ -20,13 +21,39 @@ public DateTimeOffsetGraphType()
         }
 
         /// <inheritdoc/>
-        public override object ParseLiteral(IValue value)
-            => value is StringValue stringValue ? ParseValue(stringValue.Value) : null;
+        public override object ParseLiteral(IValue value) => value switch
+        {
+            StringValue stringValue => ParseDate(stringValue.Value),
+            NullValue _ => null,
+            _ => ThrowLiteralConversionError(value)
+        };
 
         /// <inheritdoc/>
-        public override object ParseValue(object value) => ValueConverter.ConvertTo(value, typeof(DateTimeOffset));
+        public override object ParseValue(object value) => value switch
+        {
+            DateTimeOffset _ => value,
+            string s => ParseDate(s),
+            null => null,
+            _ => ThrowValueConversionError(value)
+        };
+
+        private static DateTimeOffset ParseDate(string stringValue)
+        {
+            // ISO-8601 format
+            if (DateTimeOffset.TryParseExact(stringValue, "yyyy'-'MM'-'dd'T'HH':'mm':'ss.FFFFFFFK", DateTimeFormatInfo.InvariantInfo, DateTimeStyles.AssumeUniversal | DateTimeStyles.AdjustToUniversal, out var date))
```

Comment:
Also note that we now require a ISO-8601 compliant date format to deserialize the data, rather than simply TryParse